### DISCUSSION TITLE
How to cross-compile / build for Android?

### DISCUSSION THREAD

**Discussion by kwende** [mono/mono#21500] [Q&A]
I'm trying to find resources on compiling Mono for Android, but I'm not able to find anything that works end-to-end. I see references to Scratchbox (https://wiki.maemo.org/Documentation/Maemo_5_Final_SDK_Installation#Installing_Maemo_5_SDK_using_Text-based_Installer), but the Debian packages fail to install properly. [This documentation](https://www.mono-project.com/docs/about-mono/supported-platforms/arm/) seems to indicate it is the only solution.  But yet the instructions are stale and don't work. 

Someone out there must have a build environment set up for this. Does anyone know how to do this? I have a couple of potential bugs (https://github.com/xamarin/Xamarin.Forms/issues/15388) that I would like to help solve (NOTE: this issue might be mono-related, but it's currently written up under Xamarin Forms).

**Comment by tadgem** [comment]
+1 would be good to have some more up to date info on compiling for android.

**Comment by lextm** [comment]
"Mono for Android" is the wrong keyword to use if you want to search for useful information, because that was the initial name of the product, which later changed to "Xamarin.Android". You might find releases such as https://github.com/xamarin/xamarin-android/releases/tag/v13.2.2.0 useful but right now Xamarin bits are deprecated and will not be updated.

You will have to migrate to .NET MAUI instead.

**Reply by 696558586537** [reply]
London
